lib/puppet/parser/functions/ensure_php_packages.rb

```diff
@@ -0,0 +1,33 @@
+module Puppet::Parser::Functions
+  newfunction(:ensure_php_packages, :type => :statement) do |arguments|
+    if arguments.size > 3 or arguments.size == 0
+      raise(Puppet::ParseError, "ensure_php_packages(): Wrong number of arguments given (#{arguments.size} for 2 or 3)")
+    elsif arguments.size == 3 and !arguments[2].is_a?(Hash)
```

Comment:
We should probably stick with && and || generally, unless the lower precedence is required for something.

Reply:
fair enough, will do